Pinpoint SMS sender ID not controlled by terraform

In order to configure the SMS sender ID in Pinpoint App I've created "aws_pinpoint_sms_channel" resource and set the sender_id param.
After the app was created it didn't changed the sender ID and set it as the AWS default value.

Please assist.

Looks like this is showing up in the resource acceptance testing here on update:

```
=== CONT  TestAccAWSPinpointSMSChannel_full
resource_aws_pinpoint_sms_channel_test.go:61: Step 3/3 error: Check failed: Check 2/6 error: aws_pinpoint_sms_channel.test_sms_channel: Attribute 'sender_id' expected "1234", got ""
--- FAIL: TestAccAWSPinpointSMSChannel_full (69.48s)
```

The code is only passing `sender_id` if it has a change on update:

https://github.com/hashicorp/terraform-provider-aws/blob/075f1302acb2d325c47e0dcebc169c7d544de47c/aws/resource_aws_pinpoint_sms_channel.go#L62-L64

Which seems incorrect as evidenced by the test -- the `UpdateSmsSettings` API call does not have the parameter and immediately after the `GetSmsSettings` API call has the `SenderId` unexpectedly removed.

--- a/.changelog/18281.txt
+++ b/.changelog/18281.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_pinpoint_sms_channel: Set all params on update
+```

--- a/aws/resource_aws_pinpoint_sms_channel.go
+++ b/aws/resource_aws_pinpoint_sms_channel.go
@@ -55,16 +55,16 @@ func resourceAwsPinpointSMSChannelUpsert(d *schema.ResourceData, meta interface{
 
 	applicationId := d.Get("application_id").(string)
 
-	params := &pinpoint.SMSChannelRequest{}
-
-	params.Enabled = aws.Bool(d.Get("enabled").(bool))
-
-	if d.HasChange("sender_id") {
-		params.SenderId = aws.String(d.Get("sender_id").(string))
+	params := &pinpoint.SMSChannelRequest{
+		Enabled: aws.Bool(d.Get("enabled").(bool)),
 	}
 
-	if d.HasChange("short_code") {
-		params.ShortCode = aws.String(d.Get("short_code").(string))
+	if v, ok := d.GetOk("sender_id"); ok {
+		params.SenderId = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("short_code"); ok {
+		params.ShortCode = aws.String(v.(string))
 	}
 
 	req := pinpoint.UpdateSmsChannelInput{
@@ -74,7 +74,7 @@ func resourceAwsPinpointSMSChannelUpsert(d *schema.ResourceData, meta interface{
 
 	_, err := conn.UpdateSmsChannel(&req)
 	if err != nil {
-		return fmt.Errorf("error putting Pinpoint SMS Channel for application %s: %s", applicationId, err)
+		return fmt.Errorf("error putting Pinpoint SMS Channel for application %s: %w", applicationId, err)
 	}
 
 	d.SetId(applicationId)
@@ -97,15 +97,16 @@ func resourceAwsPinpointSMSChannelRead(d *schema.ResourceData, meta interface{})
 			return nil
 		}
 
-		return fmt.Errorf("error getting Pinpoint SMS Channel for application %s: %s", d.Id(), err)
+		return fmt.Errorf("error getting Pinpoint SMS Channel for application %s: %w", d.Id(), err)
 	}
 
-	d.Set("application_id", output.SMSChannelResponse.ApplicationId)
-	d.Set("enabled", output.SMSChannelResponse.Enabled)
-	d.Set("sender_id", output.SMSChannelResponse.SenderId)
-	d.Set("short_code", output.SMSChannelResponse.ShortCode)
-	d.Set("promotional_messages_per_second", aws.Int64Value(output.SMSChannelResponse.PromotionalMessagesPerSecond))
-	d.Set("transactional_messages_per_second", aws.Int64Value(output.SMSChannelResponse.TransactionalMessagesPerSecond))
+	res := output.SMSChannelResponse
+	d.Set("application_id", res.ApplicationId)
+	d.Set("enabled", res.Enabled)
+	d.Set("sender_id", res.SenderId)
+	d.Set("short_code", res.ShortCode)
+	d.Set("promotional_messages_per_second", aws.Int64Value(res.PromotionalMessagesPerSecond))
+	d.Set("transactional_messages_per_second", aws.Int64Value(res.TransactionalMessagesPerSecond))
 	return nil
 }
 
@@ -122,7 +123,7 @@ func resourceAwsPinpointSMSChannelDelete(d *schema.ResourceData, meta interface{
 	}
 
 	if err != nil {
-		return fmt.Errorf("error deleting Pinpoint SMS Channel for application %s: %s", d.Id(), err)
+		return fmt.Errorf("error deleting Pinpoint SMS Channel for application %s: %w", d.Id(), err)
 	}
 	return nil
 }

--- a/aws/resource_aws_pinpoint_sms_channel_test.go
+++ b/aws/resource_aws_pinpoint_sms_channel_test.go
@@ -12,7 +12,7 @@ import (
 
 func TestAccAWSPinpointSMSChannel_basic(t *testing.T) {
 	var channel pinpoint.SMSChannelResponse
-	resourceName := "aws_pinpoint_sms_channel.test_sms_channel"
+	resourceName := "aws_pinpoint_sms_channel.test"
 
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:      func() { testAccPreCheck(t); testAccPreCheckAWSPinpointApp(t) },
@@ -55,7 +55,7 @@ func TestAccAWSPinpointSMSChannel_basic(t *testing.T) {
 
 func TestAccAWSPinpointSMSChannel_full(t *testing.T) {
 	var channel pinpoint.SMSChannelResponse
-	resourceName := "aws_pinpoint_sms_channel.test_sms_channel"
+	resourceName := "aws_pinpoint_sms_channel.test"
 	senderId := "1234"
 	shortCode := "5678"
 	newShortCode := "7890"
@@ -107,6 +107,29 @@ func TestAccAWSPinpointSMSChannel_full(t *testing.T) {
 	})
 }
 
+func TestAccAWSPinpointSMSChannel_disappears(t *testing.T) {
+	var channel pinpoint.SMSChannelResponse
+	resourceName := "aws_pinpoint_sms_channel.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t); testAccPreCheckAWSPinpointApp(t) },
+		ErrorCheck:    testAccErrorCheck(t, pinpoint.EndpointsID),
+		IDRefreshName: resourceName,
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSPinpointSMSChannelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSPinpointSMSChannelConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSPinpointSMSChannelExists(resourceName, &channel),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsPinpointSMSChannel(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
 func testAccCheckAWSPinpointSMSChannelExists(n string, channel *pinpoint.SMSChannelResponse) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		rs, ok := s.RootModule().Resources[n]
@@ -139,7 +162,7 @@ func testAccCheckAWSPinpointSMSChannelExists(n string, channel *pinpoint.SMSChan
 const testAccAWSPinpointSMSChannelConfig_basic = `
 resource "aws_pinpoint_app" "test_app" {}
 
-resource "aws_pinpoint_sms_channel" "test_sms_channel" {
+resource "aws_pinpoint_sms_channel" "test" {
   application_id = aws_pinpoint_app.test_app.application_id
 }
 `
@@ -148,7 +171,7 @@ func testAccAWSPinpointSMSChannelConfig_full(senderId, shortCode string) string 
 	return fmt.Sprintf(`
 resource "aws_pinpoint_app" "test_app" {}
 
-resource "aws_pinpoint_sms_channel" "test_sms_channel" {
+resource "aws_pinpoint_sms_channel" "test" {
   application_id = aws_pinpoint_app.test_app.application_id
   enabled        = "false"
   sender_id      = "%s"

